webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js

```diff
@@ -390,6 +397,27 @@ export class SakaiRubric extends RubricsElement {
       this.cloneRubric(e);
     }
   }
+
+  // SAK-47640 - Get the maximum and minimum possible grade by multiplying the max or min rating by the criterion weight.
+  getMaxPoints(criterions) {
+    return this.getPoints(criterions, Math.max);
+  }
+
+  getMinPoints(criterions) {
+    return this.getPoints(criterions, Math.min);
+  }
+
+  getPoints(criterions, minOrMax) {
+    let totalPoints = 0;
+    criterions
+      .filter( (criterion) => criterion.ratings.length > 0)
```

Comment:
I wouldn't bother with this filter, personally. This causes a pass over the whole array and I think you may as well just let the map do its thing.